pkg/osbuild/fips.go

```diff
@@ -0,0 +1,56 @@
+package osbuild
+
+import (
+	"github.com/osbuild/images/internal/common"
+	"github.com/osbuild/images/internal/fsnode"
+	"github.com/osbuild/images/pkg/disk"
+	"os"
+)
+
+func GenFIPSKernelOptions(pt *disk.PartitionTable) []string {
+	cmdline := make([]string, 0)
+	cmdline = append(cmdline, "fips=1")
+
+	genOptions := func(e disk.Entity, path []disk.Entity) error {
+		switch ent := e.(type) {
+		case *disk.Filesystem:
+			if ent.Mountpoint == "/boot" {
+				if label := ent.Label; label != "" {
+					karg := "boot=LABEL=" + label
+					cmdline = append(cmdline, karg)
+				} else if uuid := ent.UUID; uuid != "" {
+					karg := "boot=UID=" + uuid
+					cmdline = append(cmdline, karg)
+				}
+			}
+		}
+		return nil
+	}
+
+	_ = pt.ForEachEntity(genOptions)
+	return cmdline
+}
+
+func GenFIPSFiles() (files []*fsnode.File) {
+	file, _ := fsnode.NewFile("/etc/system-fips", common.ToPtr(os.FileMode(0644)),
+		"root", "root", []byte("# FIPS module installation complete\n"))
+	files = append(files, file)
+	return
+}
```

Comment:
What's the purpose of this file?  I looked around a bit and it seems to be more of a signal and not really functional.  Also it seems it's sometimes in the initrd?
Either way: If this is part of the spec, or required somehow, and if it's always done, should it be part of the stage in osbuild?

Reply:
This file is required for `fips-mode-setup --check` to return
```
 FIPS mode is enabled
```
instead of:
```
Installation of FIPS modules is not completed.
FIPS mode is enabled.
Inconsistent state detected.
```
It's not really functional indeed, but we need it at least for RHEL Edge (I think fedora does not check its existence anymore)

Reply:
I moved this part to the `GenFIPSStages()` function. Please take a look.

Reply:
So my original thought was adding this to the stage in osbuild but I realise now that that was wrong, because the stage in osbuild isn't FIPS-specific, it just runs `update-crypto-policies`.

Now that said, the change here actually breaks the feature and makes the manifest ubuildable, because it adds the stage but not the inline source which defines the content.  This part should be reverted to the earlier version which added the file to the pipeline using `p.Files = append(p.Files, osbuild.GenFIPSFiles()...)`.

Reply:
Reverted to previous implementation

Reply:
@thozza let's get it merged and we can rethink this in a followup if necessary.